MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/analytics/AnalyticsTracker.java

```diff
@@ -0,0 +1,134 @@
+package com.mapbox.mapboxandroiddemo.analytics;
+
+
+import android.os.Build;
+import android.support.annotation.NonNull;
+
+import com.segment.analytics.Analytics;
+import com.segment.analytics.messages.IdentifyMessage;
+import com.segment.analytics.messages.ScreenMessage;
+import com.segment.analytics.messages.TrackMessage;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Created by langstonsmith on 4/25/17.
+ */
+
+public class AnalyticsTracker {
+
+  private static final AnalyticsTracker INSTANCE = new AnalyticsTracker();
+
+  private String openedApp = "Opened app";
+  private String clickedOnSignInButtonEventName = "Clicked on sign in button";
+  private String clickedOnCreateAccountButtonEventName = "Clicked on create account button";
+  private String clickedOnNavDrawerSectionEventName = "Clicked on nav drawer section";
+  private String clickedOnIndividualExampleEventName = "Clicked on individual example";
+  private String sectionNameMapKey = "section name";
+  private String exampleNameMapKey = "example name";
+  private String isTabletMapValue = "tablet";
+  private String isPhoneMapValue = "phone";
+
+
+  public static AnalyticsTracker get() {
```

Comment:
Thanks @danesfeder . Thoughts @tobrun ?  I believe this is the final unresolved change from your first round of feedback.

From what I've read about using singletons, making a one-time global instance of `AnalyticsTracker` seems to be an example of when it's ok to even consider using a singleton. 

I've currently got:
```java
public static AnalyticsTracker getInstance() {
    if (analyticsInstance == null) {  // Single check
      synchronized (AnalyticsTracker.class) {
        if (analyticsInstance == null) {  // Double check
          analyticsInstance = new AnalyticsTracker();
        }
      }
    }
    return analyticsInstance;
  }
```